### PR TITLE
fix(warnings): eliminate compile warnings, dead code and typos

### PR DESCRIPTION
**Pull request by slodki**
This change is [Reviewable](https://reviewable.io/reviews/moneymanagerex/moneymanagerex/1544)

### CODE REVIEW

File: src/platfdep.h [hunk 1]
```diff
@@ -34,7 +34,6 @@ namespace mmex
     const wxFileName GetSharedDir();
     const wxFileName GetUserDir(bool create);
     const wxFileName GetLogDir(bool create);
-    static const wxFileName GetInstallDir();
```
Comment (stef145g) [inline src/platfdep.h]: This line (which has been removed) is required for a successful build.

**Comment by stef145g** [comment]
Checked conflict by switching to fix_warnings branch and rebasing onto master. Conflicting file is not existent in master causing the conflict.

File: src/platfdep.h (hunk 1 above)
Comment (slodki) [reply src/platfdep.h]: We have implemented this function in Windows only and it's must be static and not public.

Comment (stef145g) [reply src/platfdep.h]: I see that fix provided in calling function which corrects the build. Now OK

**Pull request merged**